addons/survey/static/src/question_page/question_page_list_renderer.js

```diff
@@ -51,28 +51,28 @@ export class QuestionPageListRenderer extends ListRenderer {
     }
 
     getSectionColumns(columns) {
-        const sectionColumns = [];
-
         let titleColumnIndex = 0;
-        for (const col of columns) {
-            if (col.type !== "field") {
+        let found = false;
+        let colspan = 1
+        for (let index = 0; index < columns.length; index++) {
```

Comment:
I'm still unsure what we are doing here.

If it's related to studio changes, did you check other implementations of the "getSectionColumns" method?
Survey is not the only one adding it.

Would also check earlier stable to see if the issue occurs.

If we end up really wanting this, then it will maybe require additional testing / tours?
As this kind of lines is very confusing:
`const sectionColumns = columns.slice(0, titleColumnIndex + 1).concat(columns.slice(titleColumnIndex + colspan));`.

Slicing based on adding an index and a colspan -> :exploding_head:

Reply:
Just rechecked for everything to be sure. 
So for `getSectionColumns`, the other modules that call it have only one column for a section, that's the title, and the handle for the sequence. 
In the case of Survey this is different because we have the title of the section and the number of questions when using the random selection as editable columns, plus the handle.

We do not need to worry about the handle for the computation since it doesn't need a big space to work properly

So we have not one but two columns, which needs the computation of `colspan` to be more specific and take into account the position of the column.
This is not the case for the other modules, that's why it is done only in survey, but if we want to ensure this behavior for any and all modules we would have to modify the `focusCell` inside of the `web` module.
And as a fix I don't know if that would be a good move.

As for earlier versions, since this change is made specifically for the new ListView that was added in v16, I will need to create a specific fix for the earlier versions, but from what I tested: 
1. bug happens in 14.0
2. fixed in 15.2
3. comes back in 16.0 with the new ListView

So I don't know if it's possible to skip versions to propagate the fix ? Or I can see what changed in 15.2 and propagate its fix to earlier versions, which is weird that it wasn't done in the first place.

Reply:
Not sure it's possible to skip versions but as the fix will probably be very different in 14.0 and in 16.0, we can create 2 PRs and make the first one "up to 15.0".

As this is taking bigger proportions, would you mind creating a different task for this issue?
That way we can ask proper validation by the JS framework and we avoid blocking the rest of this task.

Thanks!

Reply:
A test would be most welcomed of course